app/lib/password-generator.ts

```diff
@@ -4,10 +4,17 @@ interface PasswordOptions {
   special?: boolean;
 }
 
-export function generateRandomPassword(
+export async function generateRandomPassword(
   length = 24,
   options: PasswordOptions = { letters: true, numbers: true, special: true },
-): string {
+): Promise<string> {
+  if (window.getMockPassword) {
```

Comment:
I did this so that random password generator can be mocked in tests. Seems ok but wanted to draw your attention to this to see if you see any potential issues